### PR TITLE
Integration test with external ES cluster

### PR DESCRIPTION
**Pull request by dai-chen**
*Issue #, if available:* https://github.com/opendistro-for-elasticsearch/sql/issues/353

*Description of changes:* To automate the sanity test for new release, we need the capability to run all our integration tests with external cluster setup by release candidate artifacts, such as docker, rpm, zip etc. To make this happen, this PR covered the following code changes:

 1. **Migrating to `ESRestTestCase`**: we used `ESIntegTestCase` which has problem with external cluster.
 2. **Removing index type**: we used deprecated index type which is going to be removed in ES 8 and will be rejected by external cluster.
 3. **Fixing broken tests**: some ES API in use is actually deprecated and previous internal cluster didn't complain.

*For reviewers:* Since this is quite a large PR (all test index, data and queries using type need to be changed), **please focus on `SQLIntegTestCase` only for your convenience**.

*Testing*:

 1. All tests: `./gradlew build`
 2. Test with internal cluster: `./gradlew integTestRunner`
 3. Test with external cluster: `./gradlew integTestRunner -Dtests.cluster= -Dtests.rest.cluster=127.0.0.1:9200` (not sure why tests.cluster is required though it is not in use by new ESIntegTestCase)
 4. Comparison test: `./gradlew integTestRunner -DtestType=comparison`
 5. Doc test: `./gradlew integTestRunner -DtestType=doctest`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/MetaDataQueriesIT.java [hunk 1]
```diff
@@ -256,7 +256,7 @@ public void showWildcardIndex() throws IOException {
 
         String pattern = String.format("%s.*", TestsConstants.TEST_INDEX);
         JSONArray dataRows = getDataRows(response);
-        assertThat(dataRows.length(), equalTo(3));
+        assertThat(dataRows.length(), greaterThan(0));
```
Comment (abbashus) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/MetaDataQueriesIT.java]: What is the reason to change this? Aren't we loosing the assertion.

Comment (dai-chen) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/MetaDataQueriesIT.java]: Good catch! I didn't figure out how to do index clean up after each set of class with the new base class. The cleanup works now so I'm reverting this line of change. Thanks!

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/query/maker/AggMaker.java [hunk 2]
```diff
@@ -555,6 +555,9 @@ private DateHistogramAggregationBuilder dateHistogram(MethodField field) throws
                     case "interval":
                         dateHistogram.dateHistogramInterval(new DateHistogramInterval(kv.value.toString()));
                         break;
+                    case "fixed_interval":
+                        dateHistogram.fixedInterval(new DateHistogramInterval(kv.value.toString()));
+                        break;
```
Comment (abbashus) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/query/maker/AggMaker.java]: What is this change about?

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java [hunk 3]
```diff
@@ -530,9 +536,14 @@ private static String quoteParams(String... params) {
         }
     }
 
+    /** Explicitly pass in name used to generate variable ID because methodName is not always valid */
+    private Tuple<String, String> dateFunctionTemplate(String name, String methodName, SQLExpr field) {
+        String id = nextId(name);
+        return new Tuple<>(id, def(id, doc(field) + ".value." + methodName));
+    }
+
```
Comment (abbashus) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Sorry, not very clear about this change. Can you give an example how this affects things.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/query/maker/AggMaker.java (hunk 2 above)
Comment (dai-chen) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/query/maker/AggMaker.java]: Changes regarding ES and painless API are due to the old ones we used are deprecated. The previous testing with internal cluster didn't complain about this but it became problem now with external cluster.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java (hunk 3 above)
Comment (dai-chen) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Sorry for confusion. Basically this is caused by the painless API changes above, for example:

```
functionStr = dateFunctionTemplate("weekOfWeekyear",
                                   "get(WeekFields.ISO.weekOfWeekBasedYear())",
                                   (SQLExpr) paramers.get(0).value);
```

The old `dateFunctionTemplate(methodName, field)` passes `get(WeekFields.ISO.weekOfWeekBasedYear())` to `nextId()` which generates an invalid variable name in painless script. Will add a little more elaboration in JavaDoc.

Comment (abbashus) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Thanks, clear now.

Comment (dai-chen) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Cool. Thanks!

**Pull request merged**